tests/firebird/firebird_integration_test.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firebird
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/googleapis/genai-toolbox/internal/testutils"
+	"github.com/googleapis/genai-toolbox/tests"
+	_ "github.com/nakagami/firebirdsql"
+)
+
+var (
+	FirebirdSourceKind = "firebird"
+	FirebirdToolKind   = "firebird-sql"
+	FirebirdDatabase   = os.Getenv("FIREBIRD_DATABASE")
+	FirebirdHost       = os.Getenv("FIREBIRD_HOST")
+	FirebirdPort       = os.Getenv("FIREBIRD_PORT")
+	FirebirdUser       = os.Getenv("FIREBIRD_USER")
+	FirebirdPass       = os.Getenv("FIREBIRD_PASS")
+)
+
+func getFirebirdVars(t *testing.T) map[string]any {
+	t.Helper()
+	switch "" {
+	case FirebirdDatabase:
+		t.Fatal("'FIREBIRD_DATABASE' not set")
+	case FirebirdHost:
+		t.Fatal("'FIREBIRD_HOST' not set")
+	case FirebirdPort:
+		t.Fatal("'FIREBIRD_PORT' not set")
+	case FirebirdUser:
+		t.Fatal("'FIREBIRD_USER' not set")
+	case FirebirdPass:
+		t.Fatal("'FIREBIRD_PASS' not set")
+	}
+
+	return map[string]any{
+		"kind":     FirebirdSourceKind,
+		"host":     FirebirdHost,
+		"port":     FirebirdPort,
+		"database": FirebirdDatabase,
+		"user":     FirebirdUser,
+		"password": FirebirdPass,
+	}
+}
+
+func initFirebirdConnection(host, port, user, pass, dbname string) (*sql.DB, error) {
+	dsn := fmt.Sprintf("%s:%s@%s:%s/%s", user, pass, host, port, dbname)
+	db, err := sql.Open("firebirdsql", dsn)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create connection pool: %w", err)
+	}
+	return db, nil
+}
+
+func TestFirebirdToolEndpoints(t *testing.T) {
+	sourceConfig := getFirebirdVars(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	var args []string
+
+	db, err := initFirebirdConnection(FirebirdHost, FirebirdPort, FirebirdUser, FirebirdPass, FirebirdDatabase)
+	if err != nil {
+		t.Fatalf("unable to create firebird connection pool: %s", err)
+	}
+	defer db.Close()
+
+	shortUUID := strings.ReplaceAll(uuid.New().String(), "-", "")[:8]
+	tableNameParam := fmt.Sprintf("PARAM_TABLE_%s", shortUUID)
+	tableNameAuth := fmt.Sprintf("AUTH_TABLE_%s", shortUUID)
+	tableNameTemplateParam := fmt.Sprintf("TPARAM_TABLE_%s", shortUUID)
```

Comment:
```suggestion
	tableNameParam := fmt.Sprintf("param_table_%s", shortUUID)
	tableNameAuth := fmt.Sprintf("auth_table_%s", shortUUID)
	tableNameTemplateParam := fmt.Sprintf("template_param_table_%s", shortUUID)
```
Just to keep consistency with the other tests